src/TestableIO.System.IO.Abstractions.TestingHelpers/MockDirectory.cs

```diff
@@ -110,7 +127,25 @@ public override IFileSystemInfo CreateSymbolicLink(string path, string pathToTar
         /// <inheritdoc />
         public override IDirectoryInfo CreateTempSubdirectory(string prefix = null)
         {
-            throw CommonExceptions.NotImplemented();
+            prefix ??= "";
+            string potentialTempDirectory;
+
+            // Perform directory name generation is a loop, just in case the randomly generated name already exists.
+            do
+            {
+                var randomChars = new StringBuilder();
```

Comment:
Can't you use `Path.GetRandomFileName()` to set the `potentialTempDirectory`?

Reply:
Done. Also removed the tests that validated the character types of the random name as we now depend on the system library and whatever it's implementation decides is a valid name.